### PR TITLE
fix: DE46615 show dialog contents even when everything disabled

### PR DESCRIPTION
**Pull request by mdulat**
https://rally1.rallydev.com/#/?detail=/defect/621291213383&fdp=true

In addition to the above defect, there are a few fixes for additional issues found when working with this missing manage grades user. I will comment inline.

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-score-editor.js [hunk 1]
```diff
@@ -253,7 +253,7 @@ class ActivityScoreEditor extends ActivityEditorMixin(SkeletonMixin(LocalizeActi
 				${canSeeGrades ? html`
 					<div id="grade-info-container">
 						<div id="divider"></div>
-						<d2l-dropdown ?disabled="${!this.hasGradeHelp && !canEditGradebookStatus}">
+						<d2l-dropdown ?disabled="${(!this.hasGradeHelp && !canEditGradebookStatus) || isGradebookStatusChanging}">
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-score-editor.js:256]: The dropdown appeared disabled but actually wasn't when you clicked on it. This disables the actual dropdown as well.

File: components/d2l-activity-editor/d2l-activity-score-editor.js [hunk 2]
```diff
@@ -501,13 +502,14 @@ class ActivityScoreEditor extends ActivityEditorMixin(SkeletonMixin(LocalizeActi
...
-		this._prefetchGradeSchemes();
+		this._associateGradeSetGradebookStatus(canCreateNewGrade ? GradebookStatus.NewGrade : GradebookStatus.NotInGradebook);
 
 		const scoring = scoringStore.get(this._scoringHref);
 		scoring.setGradeMaxPoints(null);
 		scoring.setAutoPoints(false);
-		this._associateGradeSetGradebookStatus(canCreateNewGrade ? GradebookStatus.NewGrade : GradebookStatus.NotInGradebook);
+
+		this._prefetchGradeCandidates();
+		this._prefetchGradeSchemes();
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-score-editor.js:512]: All the reordering here and above is for performance, to see updated gradebook status before waiting to fetch other stuff.

File: components/d2l-activity-editor/d2l-activity-grades/d2l-activity-grades-dialog.js [hunk 3]
```diff
@@ -101,14 +101,22 @@ class ActivityGradesDialog extends ActivityEditorWorkingCopyDialogMixin(Localize
 		const associateGrade = await associateGradeStore.fetch(associateGradeHref, this.token);
 		if (!associateGrade) return;
 
-		this._createNewRadioChecked = associateGrade.gradebookStatus === GradebookStatus.NewGrade || associateGrade.gradebookStatus === GradebookStatus.NotInGradebook;
+		const canCreateNewGrade = associateGrade.canCreateNewGrade;
+
+		this._createNewRadioChecked = associateGrade.gradebookStatus === GradebookStatus.NewGrade
+			|| (canCreateNewGrade && associateGrade.gradebookStatus === GradebookStatus.NotInGradebook);
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-grades/d2l-activity-grades-dialog.js:107]: To set the checkbox properly when the dialog opens. It should only check new grade when the user has permission to add a new grade. Also, if both buttons are disabled then neither should be checked.

File: components/d2l-activity-editor/d2l-activity-grades/d2l-activity-grades-dialog.js [hunk 4]
```diff
@@ -101,14 +101,22 @@ class ActivityGradesDialog extends ActivityEditorWorkingCopyDialogMixin(Localize
...
+
+		await associateGrade.getGradeCandidates();
 
 		if (associateGrade.gradebookStatus === GradebookStatus.NotInGradebook) {
-			await this._associateGradeSetGradebookStatus(GradebookStatus.NewGrade);
+			if (canCreateNewGrade) {
+				await this._associateGradeSetGradebookStatus(GradebookStatus.NewGrade);
+			} else if (this._hasGradeCandidates) {
+				await this._associateGradeSetGradebookStatus(GradebookStatus.ExistingGrade);
+			}
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-grades/d2l-activity-grades-dialog.js:116]: Set the gradebook status when the dialog opens (if warranted)

File: components/d2l-activity-editor/d2l-activity-grades/d2l-activity-grades-dialog.js [hunk 5]
```diff
@@ -236,23 +255,27 @@ class ActivityGradesDialog extends ActivityEditorWorkingCopyDialogMixin(Localize
...
+				// If the chosen GI belongs to a Distributed Points Grade Category, then disable the Grade Out Of input.
+				scoring.setAutoPoints(gradeCandidateCollection.selected.isAutoPoints);
+				announce(`${this.localize('grades.linkingToGradeItem', { gradeName: gradeCandidateCollection.selected.name })}`);
+				if (!canCreateNewGrade) {
+					scoring.setCanUpdateScoring(false);
+				}
 			}
-			// If the chosen GI belongs to a Distributed Points Grade Category, then disable the Grade Out Of input.
-			scoring.setAutoPoints(gradeCandidateCollection.selected.isAutoPoints);
-			announce(`${this.localize('grades.linkingToGradeItem', { gradeName: gradeCandidateCollection.selected.name })}`);
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-grades/d2l-activity-grades-dialog.js:255]: All the changes within this `_saveAssociateGrade` function are to ensure we don't try to access properties of undefined when the user clicks OK. (e.g. when both buttons are disabled then nothing should actually be done).

**Review by mostlyfabulous — APPROVED** [review]

**Pull request merged**